core/io/ip.cpp

```diff
@@ -178,13 +206,12 @@ IP::ResolverID IP::resolve_hostname_queue_item(const String &p_hostname, IP::Typ
 IP::ResolverStatus IP::get_resolve_item_status(ResolverID p_id) const {
 	ERR_FAIL_INDEX_V(p_id, IP::RESOLVER_MAX_QUERIES, IP::RESOLVER_STATUS_NONE);
 
-	MutexLock lock(resolver->mutex);
```

Comment:
I'm not sure this lock should be removed. To me, it makes sense to enter a critical section every time the queue is accessed. That also leads to something else: the `status` field may not need to be a `SafeNumeric` (formerly `volatile`).
What do you think?

Reply:
I actually removed the lock because it was a `SafeNumeric` and the queue is a fixed array so its size can't change.
I agree the status could no longer be `SafeNumeric`, but in any case the function probably does not need the mutex.

The queue has fixed size, so the elements memory pointer can't change.
Even if we lock in the function, as long as we return we are no longer locked (so the state can change between the return and whatever the usage of that returned value).

Reply:
Makes sense.